In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/assign-2-train/Y_new/eventroll1_1060.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_4395.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_6168.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_9421.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_3803.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_7119.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_9600.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_3494.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_4071.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_4964.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_5751.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_6823.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_1979.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_6144.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_8939.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_1270.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_3423.npy
/kaggle/input/assign-2-train/Y_new/eventroll1_8882.npy
/kaggle/in

In [2]:
import tensorflow as tf
import os
import numpy as np
from sklearn.model_selection import train_test_split
lst_names = os.listdir("../input/assign-2-val/X_new")
file_names = []
for m in range(len(lst_names)):
    file_names.append(lst_names[m][9:])
file_names

['3566.npy',
 '3728.npy',
 '447.npy',
 '1458.npy',
 '445.npy',
 '3519.npy',
 '3616.npy',
 '696.npy',
 '2649.npy',
 '3173.npy',
 '1190.npy',
 '1018.npy',
 '934.npy',
 '2562.npy',
 '2970.npy',
 '1424.npy',
 '435.npy',
 '2178.npy',
 '3690.npy',
 '651.npy',
 '1093.npy',
 '3822.npy',
 '3889.npy',
 '2519.npy',
 '2800.npy',
 '1812.npy',
 '1635.npy',
 '3613.npy',
 '45.npy',
 '2126.npy',
 '225.npy',
 '3803.npy',
 '2326.npy',
 '1008.npy',
 '3079.npy',
 '1170.npy',
 '216.npy',
 '3500.npy',
 '2526.npy',
 '920.npy',
 '1076.npy',
 '2179.npy',
 '1523.npy',
 '1612.npy',
 '635.npy',
 '2109.npy',
 '2435.npy',
 '769.npy',
 '3363.npy',
 '2944.npy',
 '273.npy',
 '3456.npy',
 '3289.npy',
 '1560.npy',
 '2655.npy',
 '2670.npy',
 '502.npy',
 '864.npy',
 '2842.npy',
 '1805.npy',
 '708.npy',
 '1630.npy',
 '852.npy',
 '1521.npy',
 '3775.npy',
 '2509.npy',
 '259.npy',
 '3729.npy',
 '1745.npy',
 '418.npy',
 '3493.npy',
 '3496.npy',
 '2063.npy',
 '995.npy',
 '2907.npy',
 '1754.npy',
 '1371.npy',
 '2377.npy',
 '183.n

In [3]:
x_train = []
for m in range(10000):
    z = np.load("../input/assign-2-train/X_new/melspec1_" + str(m) + ".npy")
    z = np.reshape(z,(64,1000,1))
    x_train.append(z)
for m in range(2000):
    z = np.load("../input/assign-2-val/X_new/melspec1_" + file_names[m])
    z = np.reshape(z,(64,1000,1))
    x_train.append(z)
x_train = np.array(x_train)
x_train.shape  

(12000, 64, 1000, 1)

In [4]:
y_train = []
for m in range(10000):
    y_train.append(np.load("../input/assign-2-train/Y_new/eventroll1_" + str(m) + ".npy"))
for m in range(2000):
    y_train.append(np.load("../input/assign-2-val/Y_new/eventroll1_" + file_names[m]))
y_train = np.array(y_train)
y_train.shape

(12000, 10)

In [5]:
from sklearn.utils import shuffle
x_train, y_train = shuffle(x_train, y_train)
x_train, x_valid, y_train, y_valid = train_test_split(x_train, y_train, random_state=42, test_size=0.167)

In [6]:
arr1, cls_count1 = np.unique(y_valid,axis=0,return_counts=True)
arr2, cls_count2 = np.unique(y_train,axis=0,return_counts=True)

In [7]:
print(cls_count1)
print(cls_count2)

[ 22  37 177  23 147   5  43  13 181   2   1  19   2 154   2   6   1   2
   1  19 117   3  31  17 157   2   3  12   2   1   2   1  33 161   1   5
   1   2  17   1   1   8   1  40 151   1   1   4   2   7  34   4   1  31
 202   2   7   3   5   2   4  24   2   1  15   2   1   5   4   1   2   2
   1   1   2   2   1   1   1]
[127 191 816 100 676  33 185   1   1 112 929   9   3 113   8 809  18   2
  38   6   1   1   1   1   1   2 107 721   2  14 127   1 113 786  13   9
   2   1   2  59   2  17  14   1   1 143 778   3   2  17   3   1   1  16
   1  94   1   1   2   1   1  12   2   2   1   8   1 148   4 735   3   7
   7   1   7   2   1   1  42 178   3   2  10   1   3   1   3   1   1 174
 946   4  28  10   1  13  16 137   4   1   2   3   1  68   2  16   2  19
   7   1  25   1   4   1   1   7  29   1   1   1   6   3   1  10   1   1
   3   3   1   2   1   1   1   1   1   2   1   1   1]


In [8]:
print(len(cls_count1))
print(len(cls_count2))

79
139


As we can see, while converting it to a multi class problem, there are 139 possible classes for training dataset and only 79 classes in test dataset. Also, there is a huge class imbalance as many of the classes have just 1 data point. So it would not be much useful to use this technique. For more details, please read the report file.